File: models/ambulance/AmbulanceMaintenance.js
// models/ambulance/AmbulanceMaintenance.js
import mongoose from "mongoose";

const AmbulanceMaintenanceSchema = new mongoose.Schema({
  hospitalId: { type: mongoose.Schema.Types.ObjectId, ref: "Hospital", required: true },
  branchId: { type: mongoose.Schema.Types.ObjectId, ref: "Branch", required: true },

  ambulanceId: { type: mongoose.Schema.Types.ObjectId, ref: "AmbulanceMaster", required: true },
  maintenanceDate: { type: Date, default: Date.now },
  type: { type: String, enum: ["Routine", "Repair", "Accident"], default: "Routine" },
  description: String,
  cost: Number,
  performedBy: { type: mongoose.Schema.Types.ObjectId, ref: "User" } // Staff who performed or recorded
}, { timestamps: true });

export default mongoose.model("AmbulanceMaintenance", AmbulanceMaintenanceSchema);

File: jstojv/src/main/java/com/samrat/jstojv/entity/AmbulanceMaintenance.java
package com.samrat.jstojv.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "ambulance_maintenances")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AmbulanceMaintenance extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "hospital_id", nullable = false)
    private Hospital hospital;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "branch_id", nullable = false)
    private Branch branch;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "ambulance_id", nullable = false)
    private AmbulanceMaster ambulance;

    @Column(name = "maintenance_date")
    private LocalDateTime maintenanceDate = LocalDateTime.now();

    @Enumerated(EnumType.STRING)
    private Type type = Type.Routine;

    private String description;

    private Double cost;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "performed_by")
    private User performedBy;

    public enum Type {
        Routine, Repair, Accident
    }
}
